From 48fca1b6f9b1f958bc524fa3d63f982a39a495ef Mon Sep 17 00:00:00 2001
From: Andrew Hsieh <andrewhsieh@google.com>
Date: Tue, 19 Mar 2013 10:33:45 -0700
Subject: [PATCH 1/5] Add GCC/collect2 compatibility flags

Add -use-gold, -use-mcld, and use-ld passed by GCC/collect2
both ld.bfd ld.gold ignore

Change-Id: Ia64efd9ab64e06eedd720d46ef4627fae3801392
---
 tools/llvm-mcld/llvm-mcld.cpp | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/tools/llvm-mcld/llvm-mcld.cpp b/tools/llvm-mcld/llvm-mcld.cpp
index 92dec92..7a3a0c4 100644
--- a/tools/llvm-mcld/llvm-mcld.cpp
+++ b/tools/llvm-mcld/llvm-mcld.cpp
@@ -799,6 +799,23 @@ ArgTextSegAddr("Ttext",
                cl::init(-1U));
 
 //===----------------------------------------------------------------------===//
+// Ignored GCC Compatibility Options
+//===----------------------------------------------------------------------===//
+static cl::opt<bool>
+Arguse_gold("use-gold",
+          cl::desc("GCC/collect2 compatibility: uses ld.gold.  Ignored"),
+          cl::init(false));
+
+static cl::opt<bool>
+Arguse_mcld("use-mcld",
+          cl::desc("GCC/collect2 compatibility: uses ld.mcld.  Ignored"),
+          cl::init(false));
+
+static cl::opt<bool>
+Arguse_ld("use-ld",
+          cl::desc("GCC/collect2 compatibility: uses ld.bfd.  Ignored"),
+          cl::init(false));
+
 // non-member functions
 //===----------------------------------------------------------------------===//
 /// GetOutputStream - get the output stream.
-- 
1.8.1.3

